vxyowsup/tests/test_whatsapp.py

```diff
@@ -127,8 +161,7 @@ def send(self, data):
         send to lower (no lower in this layer)
         '''
         reactor.callFromThread(self.data_received.put, data)
-        ack = AckProtocolEntity(_id=data['id'], _class='message')
-        self.receive(ack.toProtocolTreeNode())
+        self.send_ack(data)
```

Comment:
Ah, definitely not. I didn't realise this was still there. This might explain some of the quirks of my code as it currently is.